src/test/socket/getsockname/test_getsockname.rs

```diff
@@ -0,0 +1,713 @@
+/*
+ * The Shadow Simulator
+ * See LICENSE for licensing information
+ */
+
+struct GetsocknameArguments {
+    fd: libc::c_int,
+    addr: Option<libc::sockaddr_in>, // if None, a null pointer should be used
+    addr_len: Option<libc::socklen_t>, // if None, a null pointer should be used
+}
+
+/// A boxed function to run as a test.
+type TestFn = Box<dyn Fn() -> Result<(), String>>;
+
+trait AsPtr<T> {
+    fn as_ptr(&self) -> *const T;
+}
+
+impl<T> AsPtr<T> for Option<T> {
+    fn as_ptr(&self) -> *const T {
+        match self {
+            Some(v) => v,
```

Comment:
Rust allows automatic type coercion from a ref to ptr as long as the type doesn't change (otherwise you need to explicitly add `as *const _`). For example Rust will automatically convert `&T` to `*const T`. For the references, Rust allows us to write it without the explicit `ref` as a convenience feature: https://github.com/rust-lang/rfcs/blob/master/text/2005-match-ergonomics.md#detailed-design (like in the first example)

I'm happy with making things explicit if it helps readability, so I will add this.